### PR TITLE
 Raise error if maximum prefix expansions are reached

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

File: src/config.h [hunk 1]
```diff
@@ -149,7 +154,7 @@ sds RSConfig_GetInfoString(const RSConfig *config);
 #define RS_DEFAULT_CONFIG                                                                       \
   {                                                                                             \
     .concurrentMode = 1, .extLoad = NULL, .enableGC = 1, .minTermPrefix = 2,                    \
-    .maxPrefixExpansions = 200, .queryTimeoutMS = 500, .timeoutPolicy = TimeoutPolicy_Return,   \
+    .maxPrefixExpansions = 800, .queryTimeoutMS = 500, .timeoutPolicy = TimeoutPolicy_Return,   \
```
Comment (MeirShpilraien) [inline src/config.h]: why you did not add prefixExpansionNoError default value here?

**Comment by mnunberg** [comment]
The change in behavior will be somewhat intentional. I guess the idea is to raise the limit high enough so that it won't matter, or to simply set the config value.

**Comment by MeirShpilraien** [comment]
I think we should not change default behavior but lets discuss it on weekly

File: src/config.h (hunk 1 above)
Comment (mnunberg) [reply src/config.h]: Because we didn't merge that other PR yet :)

**Comment by ashtul** [comment]
@gkorland @MeirShpilraien @rafie 
Doesn't it make more sense to log it rather than raise an error? 
Blocking it will prevent the issue we saw with #1578

**Comment by ashtul** [comment]
This was patched in a different way

**Pull request closed without merging**